### PR TITLE
Implemented search test with array of parameters as input data.

### PR DESCRIPTION
**Pull request by andkochergin**

### CODE REVIEW

File: spec/features/custom_filter_search_spec.rb [hunk 1]
```diff
@@ -6,19 +6,19 @@
...
+      visit search_data[:category_href]
 
-    expect(self).to have_selector(:xpath, '(//div[@class="schema-product'\
-           '__title"])[1]/a/span[contains(text(), "Samsung")]')
-    expect(all('.schema-product__title',
-               text: /^((?!Samsung).)*$/).size).to eq(0)
-    expect(all('.schema-product__description',
-               text: /^((?!43").)*$/).size).to eq(0)
+      search_data[:filters].each do |filter|
+        ::PageObjects::Elements::SearchFilter.for(filter[:filter])
```
Comment (yahorzhylinski) [inline spec/features/custom_filter_search_spec.rb]:
```
::PageObjects::Elements::SearchFilter
  .for(filter[:filter])
  .set_value(filter[:value])
```

Comment (yahorzhylinski) [reply spec/features/custom_filter_search_spec.rb]: i prefer this approach

File: spec/support/regular_expression_helper.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Helpers
+  class RegularExpression
+    include RSpec::Matchers
+    include Capybara::DSL
+
+    def self.create_regular_expression(array)
```
Comment (yahorzhylinski) [inline spec/support/regular_expression_helper.rb]: For what we need it?

File: spec/features/custom_filter_search_spec.rb (hunk 1 above)
Comment (andkochergin) [reply spec/features/custom_filter_search_spec.rb]: done

File: spec/support/regular_expression_helper.rb (hunk 2 above)
Comment (andkochergin) [reply spec/support/regular_expression_helper.rb]: This method is an extraction of logic for creating a regular expression which concat all filter array elements. I’ve extracted it into utils method to reuse it in other test in future

**Pull request merged**